dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -753,51 +881,66 @@ protected Object call() throws Exception {
                     Platform.runLater(() -> {
                         //  deepStorageBrowserPresenter.logText("Delete response code: " + deleteFolderRecursivelySpectraS3Response.getStatusCode(), LogType.SUCCESS);
                         deepStorageBrowserPresenter.logText("Successfully deleted folder", LogType.SUCCESS);
-                    });
-                } catch (final IOException e) {
-                    if (e instanceof FailedRequestException) {
-                        LOG.error("Failed to delete folder", e);
-                        Platform.runLater(() -> deepStorageBrowserPresenter.logText("Failed to delete folder : " + ((FailedRequestException) e).getError().getMessage(), LogType.ERROR));
-                        ALERT.setContentText("Failed to delete a folder");
-                        ALERT.showAndWait();
-                    } else {
-                        LOG.error("Failed to delete folder", e);
-                        Platform.runLater(() -> deepStorageBrowserPresenter.logText("Failed to delete folder", LogType.ERROR));
-                        ALERT.setContentText("Failed to delete a folder");
-                        ALERT.showAndWait();
-                    }
+                        if (ds3TreeTable.getRoot() == null || ds3TreeTable.getRoot().getValue() == null) {
+                            ds3TreeTable.setRoot(ds3TreeTable.getRoot().getParent());
+                            Platform.runLater(() -> {
+                                ds3TreeTable.getSelectionModel().clearSelection();
+                                ds3PanelPresenter.getDs3PathIndicator().setText("");
+                                ds3PanelPresenter.getDs3PathIndicatorTooltip().setText("");
+                            });
+
+                        } else {
+                            ds3TreeTable.setRoot(selectedItem);
+                        }
+                        ds3TreeTable.getSelectionModel().select(selectedItem);
+                        RefreshCompleteViewWorker.refreshCompleteTreeTableView(ds3Common, workers);
+                });
+                } catch (final FailedRequestException fre) {
```

Comment:
If this catch statement is no different than the IOException catch bellow it, remove this.